tower-batch/tests/ed25519.rs

```diff
@@ -0,0 +1,173 @@
+                    }
+                })
+            }
+            BatchControl::Flush => {
+                tracing::trace!("got flush command");
+                let batch = std::mem::replace(&mut self.batch, batch::Verifier::default());
+                let _ = self.tx.send(batch.verify(thread_rng()));
+                Box::pin(async { Ok(()) })
+            }
+        }
+    }
+}
+
+impl Drop for Ed25519Verifier {
+    fn drop(&mut self) {
+        // We need to flush the current batch in case there are still any pending futures.
+        let batch = std::mem::replace(&mut self.batch, batch::Verifier::default());
+        let _ = self.tx.send(batch.verify(thread_rng()));
+    }
+}
+
+// =============== testing code ========
+
+static LOGGER_INIT: Once = Once::new();
+
+fn install_tracing() {
+    use tracing_error::ErrorLayer;
+    use tracing_subscriber::prelude::*;
+    use tracing_subscriber::{fmt, EnvFilter};
+
+    LOGGER_INIT.call_once(|| {
+        let fmt_layer = fmt::layer().with_target(false);
+        let filter_layer = EnvFilter::try_from_default_env()
+            .or_else(|_| EnvFilter::try_new("info"))
+            .unwrap();
+
+        tracing_subscriber::registry()
+            .with(filter_layer)
+            .with(fmt_layer)
+            .with(ErrorLayer::default())
+            .init();
+    })
+}
+
+async fn sign_and_verify<V>(mut verifier: V, n: usize)
+where
+    V: Service<Ed25519Item, Response = ()>,
+    <V as Service<Ed25519Item>>::Error: Into<Box<dyn std::error::Error + Send + Sync + 'static>>,
+{
+    let mut results = FuturesUnordered::new();
+    for i in 0..n {
+        let span = tracing::trace_span!("sig", i);
+        let sk = SigningKey::new(thread_rng());
+        let vk_bytes = VerificationKeyBytes::from(&sk);
+        let msg = b"BatchVerifyTest";
+        let sig = sk.sign(&msg[..]);
+
+        verifier.ready_and().await.map_err(|e| e.into()).unwrap();
+        results.push(span.in_scope(|| verifier.call((vk_bytes, sig, msg).into())))
+    }
+
+    while let Some(result) = results.next().await {
+        let result = result.map_err(|e| e.into());
+        tracing::trace!(?result);
+        assert!(result.is_ok());
+    }
+}
+
+/*
+#[tokio::test]
+async fn individual_verification_with_service_fn() {
+    let verifier = tower::service_fn(|item: Ed25519Item| {
+        // now this is actually impossible to write, oops
+        let result = VerificationKey::try_from(vk_bytes).and_then(|vk| vk.verify(&sig, msg));
+        async move { result }
+    });
+
+    sign_and_verify(verifier, 100).await;
+}
+*/
+
+#[tokio::test]
+async fn batch_flushes_on_max_items() {
+    use tokio::time::timeout;
+    install_tracing();
+
+    // Use a very long max_latency and a short timeout to check that
+    // flushing is happening based on hitting max_items.
+    let verifier = Batch::new(Ed25519Verifier::new(), 10, Duration::from_secs(1000));
+    assert!(
```

Comment:
NIT: I think we should leverage `color-eyre` in these tests the same way we do in `zebra-state`. Instead of asserting and unwrapping put `color_eyre::Result<()>` in each of these func's return types and propagate errors out.

Reply:
Maybe we could do this at the same time as the shared test code refactoring?